presto-main/src/test/java/com/facebook/presto/type/TestRowOperators.java

```diff
@@ -169,5 +187,35 @@ public void testRowEquality()
         assertFunction("test_row(TRUE, ARRAY [1]) != test_row(TRUE, ARRAY [1])", BOOLEAN, false);
         assertFunction("test_row(TRUE, ARRAY [1]) != test_row(TRUE, ARRAY [1,2])", BOOLEAN, true);
         assertFunction("test_row(1.0, ARRAY [1,2,3], test_row(2,2.0)) != test_row(1.0, ARRAY [1,2,3], test_row(1,2.0))", BOOLEAN, true);
+
+        // Cannot be applied with different field names
```

Comment:
File a Github issue that coercions between row types need to be implemented, and then remove these commented out tests